admin-ui/plugins/auth-server/redux/features/scriptSlice.js

```diff
@@ -0,0 +1,59 @@
+import reducerRegistry from 'Redux/reducers/ReducerRegistry'
+import { createSlice } from '@reduxjs/toolkit'
+
+const initialState = {
+  scripts: [],
+  script: {},
+  loading: true,
+  scriptTypes: [],
+}
+
+const scriptSlice = createSlice({
+  name: 'scripts',
+  initialState,
+  reducers: {
+    getScripts: (state) => {
+      state.loading = true
+    },
+    getScriptsResponse: (state, action) => {
+      state.loading = false
+      if (action.payload?.data) {
+        state.scripts = action.payload.data.entries || action.payload.data || []
+      } else {
+        state.scripts = []
+      }
+    },
+    getScriptsByType: (state) => {
+      state.loading = true
+    },
+    getScriptsByTypeResponse: (state, action) => {
+      state.loading = false
+      if (action.payload?.data) {
+        state.scripts = action.payload.data.entries || action.payload.data || []
+      } else {
+        state.scripts = []
+      }
+    },
+    getScript: (state) => {
+      state.loading = true
+    },
+    getScriptResponse: (state, action) => {
+      state.loading = false
+      if (action.payload?.data) {
+        state.script = action.payload.data
+      }
+    },
+  },
+})
+
+export const {
+  getScripts,
+  getScriptsResponse,
+  getScriptsByType,
+  getScriptsByTypeResponse,
+  getScript,
+  getScriptResponse,
+} = scriptSlice.actions
+
+export const { actions, reducer, state } = scriptSlice
+reducerRegistry.register('scriptReducer', reducer)
```

Comment:
Attempting to destructure and export a non-existent `state` property from a Redux Toolkit slice is bad because it violates the principle of API contract adherence by incorrectly assuming the `createSlice` function returns a `state` property, which leads to exporting an `undefined` value and breaks the module's public interface.